### PR TITLE
Wrap entire function in the check for the ENVVAR

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

If the ENVVAR is set, then the os check will be skipped entirely

### CODE REVIEW

**Comment by PromoFaux** [comment]
Diff without whitespace for easier review:
https://github.com/pi-hole/pi-hole/pull/3596/files?diff=split&w=1

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -175,58 +175,61 @@ is_command() {
...
-    if [ "$valid_os" = true ] && [ "$valid_version" = true ]; then
-        display_warning=false
-    fi
+        if [ "$display_warning" = true ]; then
+            printf "  %b %bUnsupported OS detected%b\\n" "${CROSS}" "${COL_LIGHT_RED}" "${COL_NC}"
+            printf "      https://docs.pi-hole.net/main/prerequesites/#supported-operating-systems\\n"
+            printf "\\n"
+            printf "      This check can be skipped by setting the environment variable %bPIHOLE_SKIP_OS_CHECK%b to %btrue%b\\n" "${COL_LIGHT_RED}" "${COL_NC}" "${COL_LIGHT_RED}" "${COL_NC}"
+            printf "      e.g: 'sudo PIHOLE_SKIP_OS_CHECK=true curl -sSL https://install.pi-hole.net | bash'\\n"
+            printf "      or   'sudo PIHOLE_SKIP_OS_CHECK=true pihole -up'\\n"
```
Comment (dschaper) [inline automated install/basic-install.sh:222]: These are still not quite correct. The comments from the previous PR are valid here.

**Comment by PromoFaux** [comment]
Latest commits add the following: Tested on a Ubuntu 19.10 droplet, changed to a non-root user. Took inspiration from [here](tested2)

image

Updating:

tested

Fresh install:

tested2

**Comment by PromoFaux** [comment]
@yubiuser after we've finalised what we're doing in this PR, and it is merged, feel free to PR your addition

**Comment by yubiuser** [comment]
@PromoFaux 

You mean this PR: https://github.com/pi-hole/pi-hole/pull/3597 ;-)

**Comment by PromoFaux** [comment]
I was blind.. but now I see. 

(sleep more, Adam)

**Comment by dschaper** [comment]
This just wraps the existing in a gate, correct? Is there anything else that is needed on this or is it ready?

**Comment by PromoFaux** [comment]
> Is there anything else that is needed on this or is it ready?

Ready to go. I was going to add in a check for unexpected response, but @yubiuser has that covered in another PR.

**Comment by dschaper** [comment]
Okay. `dig @ns1` works and `ENVVAR=yes sudo -E bash` works as well. I think we have everything lined up for a fix. We can address the bypass methods (flags or text files) in the hopefully near future.

**Comment by PromoFaux** [comment]
Shall I add the `@ns1` patch into this PR?

**Comment by dschaper** [comment]
If you want to, just update the PR title and any comments so we know it's more than just a gate function added.

**Comment by yubiuser** [comment]
>Shall I add the @ns1 patch into this PR?

I would not do that: users might restrict outbound DNS requests to servers they don't control. We're going to see the same amount of support requests for this than we get now for DNS servers failing to deal with the cookie.

**Comment by dschaper** [comment]
That's fine if they do. The main goal is to move the query off the faulty local resolver. FORMERR is a bad AD DNS and cookie mismatch is a local resolver that is glitching. If you already redirect 53 to another server then you won't see this problem. If you block 53 to only a certain resolver then you know why this is failing. 

Every person so far that has either FORMERR or cookie mismatch has had no problems running the `dig @ns1` fix.

**Pull request merged**